hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java

```diff
@@ -123,6 +126,28 @@ public void run() {
         }
     }
 
+    /**
+     * If requestedQueue is null, return the assigned queue for any of the active roles. If none match return the default queue
+     * If requestedQueue is not null, make sure the requestedQueue is assigned to any of this users's roles and return it if there's a match. If it's not return the default queue
+     */
+    private String chooseQueue(String requestedQueue, List<String> roles, Map<String, String> olapServerIsolatedRoles) {
+        if (requestedQueue != null) {
+            // make sure the requested queue is available for the user roles
+            for (String role: roles) {
+                if (requestedQueue.equals(olapServerIsolatedRoles.get(role))) {
+                    return requestedQueue;
+                }
+            }
+            return "default";
+        }
+        for (String role: roles) {
+            if (olapServerIsolatedRoles.get(role) != null) {
+                return olapServerIsolatedRoles.get(role);
+            }
+        }
+        return "default";
```

Comment:
Seems like `return "default"` could be the last expression in the method _iff_ `if` were followed by `else` with `for` inside. The code looks too hard to read as is now.

```
if (requestedQueue != null) {
   ...
   // no return "default";
} else {
   ...
   // no return "default";
}
return "default";
```